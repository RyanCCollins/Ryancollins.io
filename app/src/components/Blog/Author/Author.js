import React from 'react';
import Divider from '../../Divider/Divider';
import {
  Row,
  Column,
  MediaObject,
  MediaObjectSection
} from 'react-foundation';
import {
  FaGithub,
  FaTwitter
} from 'react-icons/lib/fa';
import './Author.scss';
import ReactMarkdown from 'react-markdown';
import noPic from '../../../../images/nopic.png';

const Author = ({
  author
}) => (
  <Row>
    <Column isColumn centerOnSmall small={12} large={8}>
      <div className="author-card paper__fit">
        <h1 className="section-header">About The Author</h1>
        <Divider />
        <MediaObject stackForSmall>
          <MediaObjectSection className="center-on-small">
            <img
              alt="Blog post's author"
              className="avatar-image avatar-image__small"
              src={author.avatar ? author.avatar.url : noPic}
            />
          </MediaObjectSection>
          <MediaObjectSection>
            <h4 className="author-card__name">{`${author.name.first} ${author.name.last}`}</h4>
            <ReactMarkdown
              className="author-bio-text"
              source={author.bio ? author.bio.md : 'This author has no bio.'}
            />
            {author.website ?
              <a className="muted-link" href={author.website}>{author.website}</a>
            :
              <div></div>
            }
            <div className="author-social-links">
              {author.github ?
                <a
                  className="author-social-link"
                  href={author.github.url}
                >
                  <FaGithub className="text-medium dark-grey" />
                </a>
              :
                <div></div>
              }
              {author.twitter ?
                <a
                  className="author-social-link"
                  href={author.twitter.url}
                >
                  <FaTwitter className="text-medium dark-grey" />
                </a>
              :
                <div></div>
              }
            </div>
          </MediaObjectSection>
        </MediaObject>
      </div>
    </Column>
  </Row>
);

export default Author;
