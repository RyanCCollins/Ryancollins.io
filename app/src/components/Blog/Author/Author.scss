.flexed-media-item {
  display: flex !important;
  align-items: center;
  justify-content: center;
}

.avatar-image__small {
  border-radius: 50%;
  margin: 20px;
  width: 60px !important;
  height: 60px !important;
}

.author-card__name {
  text-align: center;
  text-transform: uppercase;
  font-size: 22px;
  font-family: 'Raleway';
}

.avatar {
  display: block;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  line-height: normal;
}

.avatar-image {
  display: inline-block;
  vertical-align: middle;
  border-radius: 100%;
  box-shadow: 0 0 10px rgba(0,0,0,.5);
  padding: 4px;
  background-color: #fff;
  border: 1px solid #ddd;
  border-radius: 4px;
  -webkit-transition: all .2s ease-in-out;
  transition: all .2s ease-in-out;
  @media screen and (max-width: 39.9375em) {
    width: 25%;
    height: 100%;
  }
}

.center-stacked {
  @media screen and (max-width: 39.9375em) {
    display: flex;
    justify-content: center;
    align-items: center;
    flex-direction: column;
  }
}

.center-on-small {
  @media screen and (max-width: 39.9375em) {
    display: flex;
    align-items: center;
    justify-content: center;
    border: 1px solid white;
  }
}

.author-card {
  margin-bottom: 50px;
}

.author-bio-text p {
  color: rgba(0,0,0,.6);
  padding: 18px 0;
  text-align: left;
  font-family: 'Open Sans';
  letter-spacing: 0;
  font-weight: 400;
  font-style: normal;
}
