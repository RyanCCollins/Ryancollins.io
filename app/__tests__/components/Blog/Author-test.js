import React from 'react';
import renderer from 'react/lib/ReactTestRenderer';

import Author from '../../../src/components/Blog/Author/Author';

// needed because of this:
// https://github.com/facebook/jest/issues/1353
jest.mock('react-dom');

const mockedAuthor = {
  avatar: '',
  bio: {
    md: 'Hello World'
  },
  name: {
    first: 'Ryan',
    last: 'Collins'
  }
}

describe('<Author />', () => {
  it('renders with required props', () => {
    const component = renderer.create(
      <Author author={mockedAuthor} />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders with social links', () => {
    const mockedAuthorWithLinks = Object.assign({}, mockedAuthor, {
      website: 'http://www.ryancollins.io',
      github: 'https://github.com/ryanccollins',
      twitter: 'https://twitter.com/ryancollinsio'
    });
    const component = renderer.create(
      <Author author={mockedAuthorWithLinks} />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
