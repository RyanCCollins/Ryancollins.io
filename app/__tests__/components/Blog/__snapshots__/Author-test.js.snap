exports[`<Author /> renders with required props 1`] = `
<div
  className="row">
  <div
    centerOnSmall={true}
    className="column small-12 large-8 small-centered"
    isColumn={true}
    large={8}
    small={12}>
    <div
      className="author-card paper__fit">
      <h1
        className="section-header">
        About The Author
      </h1>
      <span
        className="divider" />
      <div
        className="media-object center-stacked flexed-media-item">
        <div
          className="media-object-section center-on-small">
          <img
            alt="Blog post\'s author"
            className="avatar-image avatar-image__small"
            src="app/images/nopic.88380b482599d7ff3212cc8578ad7f37.png" />
        </div>
        <div
          className="media-object-section">
          <h4
            className="author-card__name">
            Ryan Collins
          </h4>
          <div
            className="author-bio-text">
            <p
              literal={null}>
              Hello World
            </p>
          </div>
          <div
            className="author-bio__social-links--wrapper">
            <span
              className="author-social-links" />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Author /> renders with social links 1`] = `
<div
  className="row">
  <div
    centerOnSmall={true}
    className="column small-12 large-8 small-centered"
    isColumn={true}
    large={8}
    small={12}>
    <div
      className="author-card paper__fit">
      <h1
        className="section-header">
        About The Author
      </h1>
      <span
        className="divider" />
      <div
        className="media-object center-stacked flexed-media-item">
        <div
          className="media-object-section center-on-small">
          <img
            alt="Blog post\'s author"
            className="avatar-image avatar-image__small"
            src="app/images/nopic.88380b482599d7ff3212cc8578ad7f37.png" />
        </div>
        <div
          className="media-object-section">
          <h4
            className="author-card__name">
            Ryan Collins
          </h4>
          <div
            className="author-bio-text">
            <p
              literal={null}>
              Hello World
            </p>
          </div>
          <div
            className="author-bio__social-links--wrapper">
            <a
              className="muted-link"
              href="http://www.ryancollins.io">
              http://www.ryancollins.io
            </a>
            <span
              className="author-social-links">
              <a
                className="author-social-link"
                href={undefined}>
                <svg
                  className="text-medium dark-grey"
                  fill="currentColor"
                  fit={true}
                  height="1em"
                  preserveAspectRatio="xMidYMid meet"
                  style={
                    Object {
                      "verticalAlign": "middle"
                    }
                  }
                  viewBox="0 0 40 40"
                  width="1em">
                  <g>
                    <path
                      d="m20.1 2.9q4.7 0 8.6 2.3t6.3 6.2 2.3 8.6q0 5.6-3.3 10.1t-8.4 6.2q-0.6 0.1-0.9-0.2t-0.3-0.7q0 0 0-1.7t0-3q0-2.1-1.2-3.1 1.3-0.2 2.3-0.4t2.1-0.9 1.8-1.5 1.2-2.3 0.5-3.4q0-2.7-1.8-4.6 0.8-2-0.2-4.5-0.6-0.2-1.8 0.2t-2 1l-0.9 0.5q-2-0.6-4.3-0.6t-4.2 0.6q-0.4-0.2-1-0.6t-1.9-0.8-1.9-0.3q-1 2.5-0.1 4.5-1.8 1.9-1.8 4.6 0 1.9 0.5 3.4t1.1 2.3 1.8 1.5 2.1 0.9 2.3 0.4q-0.9 0.8-1.1 2.3-0.4 0.2-1 0.3t-1.3 0.1-1.4-0.5-1.3-1.4q-0.4-0.7-1-1.1t-1.1-0.6l-0.5 0q-0.5 0-0.6 0.1t-0.1 0.2 0.2 0.3 0.2 0.3l0.2 0.1q0.5 0.2 1 0.9t0.7 1.1l0.2 0.5q0.3 0.9 1 1.4t1.5 0.7 1.5 0.1 1.3-0.1l0.5 0q0 0.8 0 1.9t0 1.2q0 0.5-0.3 0.7t-0.9 0.2q-5.2-1.7-8.4-6.2t-3.3-10.1q0-4.7 2.3-8.6t6.2-6.2 8.6-2.3z m-10.6 24.6q0.1-0.2-0.2-0.3-0.2-0.1-0.2 0.1-0.1 0.1 0.1 0.2 0.2 0.2 0.3 0z m0.7 0.7q0.1-0.1-0.1-0.3-0.2-0.2-0.3-0.1-0.2 0.1 0 0.4 0.3 0.2 0.4 0z m0.7 1q0.2-0.1 0-0.4-0.2-0.3-0.4-0.1-0.2 0.1 0 0.4t0.4 0.1z m0.9 1q0.2-0.2-0.1-0.4-0.3-0.3-0.4-0.1-0.2 0.2 0 0.4 0.3 0.3 0.5 0.1z m1.3 0.5q0-0.2-0.3-0.3-0.4-0.1-0.4 0.1t0.2 0.4q0.4 0.1 0.5-0.2z m1.4 0.1q0-0.2-0.4-0.2-0.4 0-0.4 0.2 0 0.3 0.4 0.3 0.4 0 0.4-0.3z m1.3-0.2q-0.1-0.2-0.4-0.2-0.4 0.1-0.3 0.4t0.4 0.1 0.3-0.3z" />
                  </g>
                </svg>
              </a>
              <a
                className="author-social-link"
                href={undefined}>
                <svg
                  className="text-medium dark-grey"
                  fill="currentColor"
                  fit={true}
                  height="1em"
                  preserveAspectRatio="xMidYMid meet"
                  style={
                    Object {
                      "verticalAlign": "middle"
                    }
                  }
                  viewBox="0 0 40 40"
                  width="1em">
                  <g>
                    <path
                      d="m37.7 9.1q-1.5 2.2-3.7 3.7 0.1 0.3 0.1 1 0 2.9-0.9 5.8t-2.6 5.5-4.1 4.7-5.7 3.3-7.2 1.2q-6.1 0-11.1-3.3 0.8 0.1 1.7 0.1 5 0 9-3-2.4-0.1-4.2-1.5t-2.6-3.5q0.8 0.1 1.4 0.1 1 0 1.9-0.3-2.5-0.5-4.1-2.5t-1.7-4.6v0q1.5 0.8 3.3 0.9-1.5-1-2.4-2.6t-0.8-3.4q0-2 0.9-3.7 2.7 3.4 6.6 5.4t8.3 2.2q-0.2-0.9-0.2-1.7 0-3 2.1-5.1t5.1-2.1q3.2 0 5.3 2.3 2.4-0.5 4.6-1.7-0.8 2.5-3.2 3.9 2.1-0.2 4.2-1.1z" />
                  </g>
                </svg>
              </a>
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
